src/components/axes/timeAxis.ts

```diff
@@ -278,15 +289,40 @@ export module Axis {
       return Math.min(this._getIntervalLength(config), this.width()) >= worstWidth;
     }
 
+    protected _getSize(availableWidth: number, availableHeight: number) {
+      // Makes sure that the size it requires is a multiple of tier sizes, such that
+      // we have no leftover tiers
+
+      var size = super._getSize(availableWidth, availableHeight);
+
+      var adjustedHeight = 0;
+      for (var i = 0; i < this._tierHeights.length; i++) {
+        if (adjustedHeight + this._tierHeights[i] > size.height) {
+          break;
+        }
+        adjustedHeight += this._tierHeights[i];
+      }
```

Comment:
I think this version is the most performant, but what would happen if we were to encapsulate this into a reduce statement?